src/symbolize/symbolizer.rs

```diff
@@ -611,10 +648,26 @@ impl Symbolizer {
                             "APK symbolization does not support absolute address inputs",
                         ))
                     }
-                    Input::FileOffset(offset) => {
-                        todo!()
-                    }
-                };
+                    Input::FileOffset(offset) => match normalize_apk_offset(offset, path)? {
+                        Some((elf_addr, _elf_path, elf_parser)) => {
+                            let elf_parser = Rc::new(elf_parser);
+                            // TODO: Duplicated with `ElfCache`. Needs to be unified.
+                            #[cfg(feature = "dwarf")]
+                            let backend = ElfBackend::Dwarf(Rc::new(DwarfResolver::from_parser(
+                                elf_parser,
+                                self.elf_cache.code_info(),
+                                self.elf_cache.debug_syms(),
+                            )?));
+
+                            #[cfg(not(feature = "dwarf"))]
+                            let backend = ElfBackend::Elf(elf_parser);
+
+                            let resolver = ElfResolver::with_backend(path, backend)?;
+                            self.symbolize_with_resolver(elf_addr, &resolver)
+                        }
+                        None => return Ok(Symbolized::Unknown),
```

Comment:
this is a completely duplicate of the logic slightly higher in this commit, right? It seems big enough to warrant extracting into a common function.

Reply:
Yeah, but it's only big enough because of all the mumbo-jumbo we currently have for the resolver creation. This should boil down to a single line once the caching logic has been reworked. I don't want to introduce a function now just to remove it later.

Reply:
ok, no problem